storage/backend/ceph/internals.go

```diff
@@ -1,26 +1,69 @@
 package ceph
 
 import (
+	"encoding/json"
 	"fmt"
 	"os/exec"
 	"regexp"
 	"strings"
 
-	"encoding/json"
-
 	"github.com/contiv/volplugin/storage"
 
 	log "github.com/Sirupsen/logrus"
 )
 
+func (c *Driver) lockImage(do storage.DriverOptions) error {
+	poolName := do.Volume.Params["pool"]
+
+	if err := exec.Command("rbd", "lock", "add", do.Volume.Name, do.Volume.Name, "--pool", poolName).Run(); err != nil {
+		log.Debugf("Could not acquire lock for %q: %v", do.Volume.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *Driver) unlockImage(do storage.DriverOptions) error {
+	poolName := do.Volume.Params["pool"]
+
+	output, err := exec.Command("rbd", "lock", "--format", "json", "list", do.Volume.Name, "--pool", poolName).Output()
+	if err != nil {
+		return fmt.Errorf("Error running `rbd lock list` for volume %q: %v", do.Volume.Name, err)
+	}
+
+	locks := map[string]map[string]string{}
+
+	if err := json.Unmarshal(output, &locks); err != nil {
+		return fmt.Errorf("Error unmarshalling lock report for volume %q: %v", do.Volume.Name, err)
+	}
+
+	if _, ok := locks[do.Volume.Name]; ok {
+		if err := exec.Command("rbd", "lock", "remove", do.Volume.Name, do.Volume.Name, locks[do.Volume.Name]["locker"], "--pool", poolName).Run(); err != nil {
```

Comment:
similar to above, may be we should use `CombinedOutput()` instead of `Run()` and log the output in case of failures.